### PR TITLE
Add more tests to t/run/todo.t

### PR DESCRIPTION
**Pull request by khwilliamson**
* This set of changes does not require a perldelta entry.

### CODE REVIEW

File: t/run/todo.t [hunk 1]
```diff
@@ -36,4 +41,34 @@ TODO: {
     "undef:de567\nundef:de567", { eval $switches }, "");
 }
 
+TODO: {
+    local $::TODO = 'GH 16876';
+    fresh_perl('$_ = "a"; s{ x | (?{ s{}{x} }) }{}gx;',
+               { stderr => 'devnull' });
+    is($?, 0, "No assertion failue");
```
Comment (jkeenan) [inline t/run/todo.t]: `s/failue/failure/g` throughout

Comment (khwilliamson) [reply t/run/todo.t]: thanks

**Comment by jkeenan** [comment]
Test failed CI; please see https://github.com/Perl/perl5/pull/22680/checks

**Comment by jkeenan** [comment]
I checked out this branch, rebased on blead, built and tested.  I got this result.
```
$ cd t;./perl harness -v run/todo.t; cd -

ok 1 - GH \#16894
not ok 2 - "abcde5678" =~ / b (*pla:.*(*plb:(*plb:(.{4}))? (.... # TODO GH 16250
ok 3 - No assertion failure # TODO GH 16876
ok 4 - No assertion failure # TODO GH 16952
not ok 5 - No assertion failure # TODO GH 16971
ok 6 - No assertion failure # TODO GH 16627
ok
All tests successful.

Test Summary Report
-------------------
run/todo.t (Wstat: 0 Tests: 6 Failed: 0)
  TODO passed:   3-4, 6
Files=1, Tests=6,  0 wallclock secs ( 0.00 usr  0.00 sys +  0.02 cusr  0.01 csys =  0.03 CPU)
Result: PASS
```
Is there a better place to put tests that were once `TODO`ed but are now passing?

**Comment by khwilliamson** [comment]
There may be a better place, but I left it here to show that it could be left here.

**Comment by tonycoz** [comment]
Don't these belong in the test scripts that test that particular op? like the many other TODO tests?

**Comment by khwilliamson** [comment]
Yes, but my goal here was to make it easier for someone to just start going through tickets and copying the tests without having to think about anything more than necessary.  They can be moved later.

**Pull request merged**